### PR TITLE
Refactor for local cache

### PR DESCRIPTION
**Pull request by calvinjia**
Better division for the file system configuration and constructors.

### CODE REVIEW

File: core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java [hunk 1]
```diff
@@ -491,9 +491,13 @@ public synchronized void initialize(URI uri, org.apache.hadoop.conf.Configuratio
     // Create FileSystem for accessing Alluxio.
     // Disable URI validation for non-Alluxio schemes.
     boolean disableUriValidation =
-        (uri.getScheme() != null) ? uri.getScheme().equals(Constants.SCHEME) : true;
-    mFileSystem = FileSystem.Factory.create(
-        ClientContext.create(subject, alluxioConf).setUriValidationEnabled(disableUriValidation));
+        (uri.getScheme() == null) || uri.getScheme().equals(Constants.SCHEME);
+    if (alluxioConf.getBoolean(PropertyKey.USER_LOCAL_CACHE_LIBRARY)) {
+      mFileSystem = FileSystem.Factory.create(null, alluxioConf);
```
Comment (bf8086) [inline core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: Would it be okay now that the file system client does not have the client user subject?

File: core/client/fs/src/main/java/alluxio/client/file/FileSystem.java [hunk 2]
```diff
@@ -130,9 +130,17 @@ public static FileSystem create(ClientContext ctx) {
      * @return a new FileSystem instance
      */
     public static FileSystem create(FileSystemContext context) {
+      return create(context, context.getClusterConf());
```
Comment (bf8086) [inline core/client/fs/src/main/java/alluxio/client/file/FileSystem.java]: Would it be cleaner if we just have a subclass of `FileSystemContext` that override `getClusterConf` to return the client side configuration instead?

Comment (calvinjia) [reply core/client/fs/src/main/java/alluxio/client/file/FileSystem.java]: `FileSystemContext` wasn't written to be easily extended, and even if we had a subclass we would still need a switch somewhere to make that FileSystemContext version

File: core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java (hunk 1 above)
Comment (calvinjia) [reply core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java]: that shouldn't be necessary in library mode

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**